Tests/UITests/TestApp/TestAppDelegate.swift

```diff
@@ -14,26 +14,38 @@ class TestAppDelegate: SpeziAppDelegate {
     override var configuration: Configuration {
         Configuration(standard: HealthKitTestAppStandard()) {
             HealthKit {
-                CollectSample( //
+                CollectSample(
                     HKQuantityType.electrocardiogramType(),
                     deliverySetting: .background(.manual)
                 )
-                CollectSample( //
-                    HKQuantityType(.stepCount),
-                    deliverySetting: .background(.automatic)
-                )
                 CollectSample(
                     HKQuantityType(.pushCount),
                     deliverySetting: .anchorQuery(.manual)
                 )
-                CollectSample( //
-                    HKQuantityType(.activeEnergyBurned),
-                    deliverySetting: .anchorQuery(.automatic)
-                )
                 CollectSample( //
                     HKQuantityType(.restingHeartRate),
                     deliverySetting: .manual()
                 )
+                BulkUpload(
+                    Set([HKQuantityType(.stepCount)]),
+                    predicate: HKQuery.predicateForSamples(
+                        withStart: Calendar.current.date(byAdding: .month, value: -6, to: .now),
+                        end: nil,
```

Comment:
Re the previous comment about the BulkUpload triggering, I'm not 100% about how `end: nil` is interpreted, but we  might want to have the query end with `Date()` such that we don't double count samples? 

Couldn't find an answer in the [documentation](https://developer.apple.com/documentation/healthkit/hkquery/1614771-predicateforsamples) but it might mean the predicate includes all incoming samples in the future?

Reply:
Im actually not too sure about this either, but your point makes sense - these shouldnt be collected based on the current predicate. This is the answer I got from GPT:

"The provided query predicate is interpreted as follows:

- `HKQuery.predicateForSamples(withStart:end:options:)` is a method used to create a predicate for querying HealthKit samples within a specific time range.
- In this case:
  - `withStart` parameter specifies the start date of the query's time range. It is calculated by subtracting 6 months from the current date using `Calendar.current.date(byAdding: .month, value: -6, to: .now)`.
  - `end` parameter is set to `nil`, indicating that there is no end date specified, meaning the query will continue to the current time.
  - `options` parameter is set to `.strictEndDate`, which specifies that the end date is exclusive, meaning samples collected exactly at the end date will not be included in the query results.

So, the predicate will retrieve HealthKit samples collected within the last 6 months up to the current moment, excluding any samples collected exactly at the current time."

Would it be possible that since the samples are uploaded milliseconds (or a very very short time) before the query starts executing, they're still being collected? And since `end` is nil, it includes these since its technically not the current time?

Reply:
I think specifying `Date()` should fix it, will give it a try and also modify the samples that I inject

Reply:
Thanks for looking into this – not sure I trust GPT's answer 100% since the answer is sparsely documented on the internet, but it could be true. I think it's a good idea for the BulkUpload to end at `Date()` either way, both in this test app and our Prisma app, since that clear specifies which samples belong in BulkUpload vs CollectSamples